docs/collecting-data/code-generation/using-the-cli/index.md

```diff
@@ -0,0 +1,71 @@
+---
+title: "Working with the Snowtype CLI"
+sidebar_position: 2
+---
+
+## Working with the Snowtype CLI
+
+The Snowtype CLI is a tool which aims to speed up tracking implementations, provide type safety and inline documentation for developers and ultimately reduce the number of erroneous events. By integrating this tool in the development workflow we introduce a way to connect the additions and updates done in a Snowplow implementation with the corresponding tracking code of the project.
+
+## Authenticating with the Console
+
+It is required for the Snowtype CLI to work properly, that you have a Console API key available. You can find your own or create one in the Console [credentials page](https://console.snowplowanalytics.com/credentials).
+
+From there, the ways for the CLI to recognize the key are either through the global `-k, --apiKey` option or the `SNOWPLOW_CONSOLE_API_KEY` environment variable. The Snowtype CLI does automatically read from a `.env` file as well.
+
+**Recommended:** We recommend that you use the `SNOWPLOW_CONSOLE_API_KEY` environment variable.
+
+## Generate tracking code
+
+The CLI will generate tracking code using a valid Snowtype configuration file with the `snowtype generate command`. The code that will be generated, depending on the language, will have all the required types used in schemas and Tracking Scenarios together with methods/classes that allows for tracking these.
+
+The code generated by the CLI is not minified and contains inline documentation for methods, classes and types. If needed you can modify it in any way it suits your project.
+
+### Contents
+The contents of a generated file from the Snowtype CLI will be:
+- Types/Interfaces/Classes for each schema that relates to the Data Structures, Iglu Central Schemas and Tracking Scenarios selected.
+- For each schema:
+    - A method/class to instantiate the structure as a Self Describing JSON. _This is particularly useful to add entities as extra context on events._
+    - A method that sends a Self Describing Event with the schema as the main event entity.
```

Comment:
what does this mean?

Reply:
trackX, would send a self describing event with the X schema as the event data information.
We can consider X as the entity describing the event and the rest entities as context of the event.

Reply:
let's not add more complexity to the `context`/`entity` naming problem 🙈 We should only use "entity" to mean "context entity"